src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -1,22 +1,52 @@
-package mate.academy.hibernate.relations.dao.impl;
-
-import java.util.Optional;
-import mate.academy.hibernate.relations.dao.ActorDao;
-import mate.academy.hibernate.relations.model.Actor;
-import org.hibernate.SessionFactory;
-
-public class ActorDaoImpl extends AbstractDao implements ActorDao {
-    public ActorDaoImpl(SessionFactory sessionFactory) {
-        super(sessionFactory);
-    }
-
-    @Override
-    public Actor add(Actor actor) {
-        return null;
-    }
-
-    @Override
-    public Optional<Actor> get(Long id) {
-        return null;
-    }
-}
+package mate.academy.hibernate.relations.dao.impl;
+
+import java.util.Optional;
+import mate.academy.hibernate.relations.dao.ActorDao;
+import mate.academy.hibernate.relations.exception.DataProcessingException;
+import mate.academy.hibernate.relations.model.Actor;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+public class ActorDaoImpl extends AbstractDao implements ActorDao {
+    public ActorDaoImpl(SessionFactory sessionFactory) {
+        super(sessionFactory);
+    }
+
+    @Override
+    public Actor add(Actor actor) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(actor);
+            transaction.commit();
+        } catch (RuntimeException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add actor to DB. Actor: " + actor);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return actor;
+    }
+
+    @Override
+    public Optional<Actor> get(Long id) {
+        Optional<Actor> actor;
+        try (Session session = factory.openSession()) {
+            Transaction transaction = session.beginTransaction();
+            actor = Optional.ofNullable(session.get(Actor.class, id));
+            if (actor.isPresent()) {
+                transaction.commit();
+            }
+        } catch (RuntimeException e) {
+            throw new DataProcessingException("Can't get from DB actor by id: " + id);
+        }
+        return actor;

```

Comment:
```suggestion
        try (Session session = factory.openSession()) {
            return Optional.ofNullable(session.get(Actor.class, id));
        } catch (RuntimeException e) {
            throw new DataProcessingException("Can't get from DB actor by id: " + id);
        }
```
You can simplify it. Apply to all similar methods